src/frameworks/drupal8/solr.md

```diff
@@ -1,30 +1,112 @@
-the server a name and a description.
+The configuration can be managed from `settings.platformsh.php` by adding the following code snippet.
+
+- Edit the value of `$relationship_name` if you are using a different relationship.
+
+- Edit the value of `$solr_server_name` if you want to configure a Solr server in Drupal other than the default server automatically created by Search API Solr module.
+
+```php
+<?php
+// Configures the Solr server.
+if (isset($_ENV['PLATFORM_RELATIONSHIPS'])) {
+  $relationships = json_decode(base64_decode($_ENV['PLATFORM_RELATIONSHIPS']), TRUE);
 
--   Server class: *Solr service*
--   Solr host: `solr.internal`
--   Solr port: `8080`
--   Solr path: `/solr`
+  // Edit this value to match the Solr relationship name defined in your
+  // '.platform.app.yaml' file. In this example it is 'solr'.
+  $relationship_name = 'solr';
+
+  if (!empty($relationships[$relationship_name][0])) {
+    // Edit this value to use the the machine name of the Solr server in Drupal
+    // if you are using a different server than the default one automatically
+    // created by Search API, which is named 'default_solr_server'.
+    $solr_server_name = 'default_solr_server';
+
+    $solr = $relationships[$relationship_name][0];
+
+    // Gets the name of the Solr core from the Platform.sh relationship. Uses
+    // 'collection1' if empty to conform with the default Solr service single
+    // core configuration for versions lower than 6.x.
+    $core = substr($solr['path'], 5) ? : 'collection1';
+
+    $config['search_api.server.' . $solr_server_name]['backend_config']['connector_config']['core'] = $core;
+
+    // The path is always 'solr'.
+    $config['search_api.server.' . $solr_server_name]['backend_config']['connector_config']['path'] = 'solr';
+
+    // Gets the host and port from the values returned from the relationship.
+    $config['search_api.server.' . $solr_server_name]['backend_config']['connector_config']['host'] = $solr['host'];
+    $config['search_api.server.' . $solr_server_name]['backend_config']['connector_config']['port'] = $solr['port'];
+  }
+}
+```
 
 > **note**
-> The name of the host depends on the name you gave the relationship in `.platform.app.yaml`. If
-> you called the relationship `solrsearch` instead, then the hostname will be `solrsearch.internal`.
+> At this time, bugs in Drupal 8 configuration system prevent new Solr servers from being added via configuration. The default server is automatically created by Search API Solr module, so this configuration will work fine for it. For multiple Solr servers or a different one other than the default you must add the new server(s) at the Search API administration page `/admin/config/search/search-api/add-server`. The newly created server(s) will use its configuration from code. See [these](https://www.drupal.org/node/2682369) [issues](https://www.drupal.org/node/2744057) for more information.
```

Comment:
Strictly speaking, Search API Solr doesn't create a core on install.  There's another module it ships with that does so if you enable it and then remove it.  I don't know if that's too subtle a distinction to get into here.  Thoughts?